powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,98 @@
 import copy
 
-from powersimdata.input.grid import Grid
 from powersimdata.input.input_data import InputData
+from powersimdata.network.usa_tamu.constants.zones import interconnect2id
 
 
 class TransformProfile(object):
-    """Transforms profile according to operations listed in change table."""
+    """Transform profile according to operations listed in change table."""
 
     def __init__(self, scenario_info, grid, ct):
-        """Constructor
+        """Constructor.
 
         :param dict scenario_info: scenario information.
         :param powersimdata.input.grid.Grid grid: a Grid object.
         :param dict ct: change table.
         """
         self._input_data = InputData()
         self.scenario_info = scenario_info
-        self.grid = copy.deepcopy(grid)
+
         self.ct = copy.deepcopy(ct)
+        self.grid = copy.deepcopy(grid)
```

Comment:
Should we mention in the docstring that the grid passed in is already transformed? I was just realizing that we have that assumption built in here when dealing with new plants

Reply:
yes we should. Unfortunately, we cannot checked the type since base grid and transformed grid are both `Grid` object. It should be mentioned in the docstring at the very least.

Reply:
I suppose we could change the parameter type to `TransformGrid` and call `get_grid`, but yeah for now just updating the docstring sounds good.

Reply:
I thought about it. It is not optimal in the `Analyze` class because the grid is retrieve from the `mat` file and hence does not need to be transformed. We could get instantiate a `TransformGrid` object with an empty `dict` for the change table.

If we want to do that I would wait that we have the refactor of the `create` module first because the `TransformProfile` class is instantiated there.

Reply:
I have updated the docstring